### DISCUSSION TITLE
[@types/office-js] issues with contextual tabs

### DISCUSSION THREAD

**Discussion by lbourda** [DefinitelyTyped/DefinitelyTyped#53920] [Issues with a @types package] [answered]
1/ "enabled" works, but not on Menu controls (not talking about subitems, but the main control itself).
Moreover, if you try to enable/disable two controls at a time and one of them is a menu, none of them will change.

2/ "visible" doesn't work on controls in office-js-preview, and has been removed from office-js while fixing the "groups" implementation (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53367).
And again if you try to enable/disable a control while changing the visibility of another, none of them will change.
On the other hand, it does work on tabs.

3/ is the "shared runtime" limitation enforcing "exactly the same string as you used for the resid of the <Runtime> element" really necessary?
Someone from Microsoft advised us to use the syntax sharedurl#tab1 and sharedurl#tab2 for our two buttons opening the taskpane but on different tabs inside the pane, which works fine enough, but is therefore not compatible with the current contextual tabs limitation.

Please let me know if you need reproducing steps / github repo.

**Comment by ElizabethSamuel-MSFT** [comment]
@lbourda Apologies for our delayed response.

We have an article ([Create custom contextual tabs in Office Add-ins](https://learn.microsoft.com/office/dev/add-ins/design/contextual-tabs)) that discusses this feature. The [Resources section](https://learn.microsoft.com/office/dev/add-ins/design/contextual-tabs#resources) includes links to a related demo and GitHub sample. Can you let us know if any of this helps address your questions?

Tagging @davidchesnut to take a look.

Thanks.

**Comment by davidchesnut** [comment, marked as answer]
Hi @lbourda,

It looks like this issue is more about the behavior of the contextual tabs feature, than the package types. If you are still seeing the problems you described can you recreate this issue at https://github.com/OfficeDev/office-js/issues ? That's where we track product bugs and issues and it will be much easier to get it assigned to the right person there.

Thanks!
David
